### ISSUE TITLE
Windows 11 ntdll.dll Symbols Not Completely Loaded

### ISSUE THREAD

**Issue by gwillcox-r7** [NationalSecurityAgency/ghidra#4132]
**Describe the bug**
When loading Windows 11's copy of ntdll.dll into the latest version of Ghidra (10.1.2), not all of the symbols will be loaded, whereas using tools such as Relyze or IDA the same symbols will be completely loaded.

**To Reproduce**
1. Find a Windows 11 machine.
2. Grab the ntdll.dll file from that machine and load it into Ghidra
3. Analyze it using default settings, can also change PDB MSDIA setting and enable that if you want, should be the same regardless.
4. Wait for analysis to end.
5. Verify that functions like `_Rtlp***` specifically functions like `_RtlpMuiRegInitLIPLanguage` get labeled in Relyze and IDA in the function list, yet these don't appear in the function list in Ghidra.

**Expected behavior**
We get all the symbols loaded so that nearly all of the functions are labeled with appropriate names.

**Screenshots**
![Screenshot 2022-04-05 133106](https://user-images.githubusercontent.com/63261883/161825315-9a3c83d3-746b-4f9e-9e7a-7c0f8b9fc531.png)
![Screenshot 2022-04-05 133046](https://user-images.githubusercontent.com/63261883/161825316-f65636f9-72ee-49e7-a8d8-aef6c09db8f0.png)

**Attachments**
Sorry can't send the DLL here, but 10.0.22000.527 should be the `ntdll.dll` version.

**Environment (please complete the following information):**
 - OS: Windows 11
 - Java Version: OpenJDK 15.0.1
 - Ghidra Version: 10.1.2
 - Ghidra Origin: Downloaded from prebuilt ZIP files on this repo's Releases section.

**Comment by ghizard** [comment]
With MSDIA PDB Analyzer, we used to make the (public) mangled symbols "primary" as the (global) plain-name symbol did not carry additional information.  Being primary is what allows the demangler analyzer to do its job (in Ghidra, only one symbol at an address can be primary).  With PDB Universal, we changed this order to take advantage of the sequence of much more rich information that is available with the sequence of global symbols (that MSDIA PDB analyzer was not aware of) such as frame information, parameters, and variable scopes.  In addition, the demangler is only aware of structure and class names, not their members, but PDB global symbols can reference the "type" database within the PDB directly to give exact type definitions.  So this is usually the better solution.  However, for your PDB (like other MSFT PDBs), type information is not conveyed in the PDB.  Moreover, the sequence of globals for your function is only composed of the start/stop set of symbols and does not contain the other rich information.

There is a short-term fix that can be made for this where we just detect whether there is type information in the PDB to process accordingly.  But this change would likely only be temporary, as we have future plans for more sophisticated processing to deal with more complicated situations.

If you have ability to make a fix in the source for yourself, you can modify PdbApplicator (in Ghidra 10.1.5 or earlier) or DefaultPdbApplicator (master) to make a small fix.
At approximately lines 284-285 (DefaultPdbApplicator), you can replace
```
		processGlobalSymbolsNoTypedefs();
		processPublicSymbols();
```
with
```
		if (pdb.getTypeProgramInterface().getTypeIndexMaxExclusive() == pdb.getTypeProgramInterface().getTypeIndexMin()) {
			processPublicSymbols();
			processGlobalSymbolsNoTypedefs();
		}
		else {
			processGlobalSymbolsNoTypedefs();
			processPublicSymbols();
		}
```

**Comment by ghizard** [comment]
Hello.  I'm having difficulty replicating your results.  I first tried out of our master repository with no problems.  Then I pull our 10.1.2 distribution and also had no problems.

I imported ntdll.dll and then lauched it in the Code Browser.  Since I didn't already have the PDB, I skipped automated analysis and went straight to the File->Load PDB File... action.  But in order to get the PDB, I had to configure the Symbol Server Config to allow it to know where to try to find the PDB (I had to elect to have the MSFT symbol server in the config).  Once configured, I loaded the PDB, which kicked off some automatic analysis, and I see the _Rtlp functions in the Symbol Tree in the Functions folder structure.  I can also use the Symbol Table to filter on the _Rtlp functions.

Is it possible that you did not check the Symbol Server Config?  Did you run Automatic Analysis immediately before having the configuration done?

**Comment by gwillcox-r7** [comment]
Ah okay so I think I figured out what the issue was. When using the universal PDB loader the option "Search remote symbol servers" was not checked on my install for some odd reason. Ticking this allowed it to reach out to the symbol server and correctly update things.

My sincere apologies for the confusion here, seems that setting isn't enabled by default which was leading to some confusion.

Closing this ticket as its not an issue but instead a misunderstanding and a configuration mistake.
